src/java.base/share/classes/java/lang/constant/MethodTypeDescImpl.java

```diff
@@ -170,11 +170,17 @@ public String descriptorString() {
         if (desc != null)
             return desc;
 
-        var sj = new StringJoiner("", "(", ")" + returnType().descriptorString());
-        for (int i = 0; i < parameterCount(); i++) {
-            sj.add(parameterType(i).descriptorString());
+        int len = 2 + returnType.descriptorString().length();
+        for (ClassDesc argType : argTypes) {
+            len += argType.descriptorString().length();
         }
-        return cachedDescriptorString = sj.toString();
+        StringBuilder sb = new StringBuilder(len).append('(');
+        for (ClassDesc argType : argTypes) {
+            sb.append(argType.descriptorString());
+        }
+        desc = sb.append(')').append(returnType().descriptorString()).toString();
```

Comment:
Nit: the rest of the code (and even the new sizing part you added) uses `returnType` instead of `returnType()`

Reply:
Think the `returnType()` call was for parity with displayDescriptor changes; now that we are just in the impl class, either way is fine.

Reply:
Thanks, fixed.